app/javascript/angular/code/services/_fixed_sessions.js

```diff
@@ -95,7 +67,9 @@ export const fixedSessions = (
       if (fetchableSessionsCount) {
         this.fetchableSessionsCount = fetchableSessionsCount;
       }
-      sessionsUtils.onSessionsFetch(this);
+      if (sessionsUtils.isSessionSelected()) {
+        this.reSelectSession(sessionsUtils.selectedSessionId());
+      }
```

Comment:
that's basically what it was doing before, but I prefer if it's here directly, makes it easier to understand and gives us a fighting chance to refactor this in the future 😄